usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/ParseSingleRowForFileTest.java

```diff
@@ -75,52 +92,132 @@ void shouldValidateAndParseOneByOne() {
+        ValidationResultRepository mockResultRepo = mockResultRepository();
+        GtfsSpecRepository mockSpecRepo = mockSpecRepository();
+        RawFileRepository mockFileRepo = mockFileRepository();
 
         ParseSingleRowForFile underTest = new ParseSingleRowForFile(
                 RawFileInfo.builder().filename("test_invalid.tst").build(),
-                new MockRawFileRepo(),
-                new MockSpecRepo(),
-                resultRepo
+                mockFileRepo,
+                mockSpecRepo,
+                mockResultRepo
         );
 
         underTest.execute();
-        assertEquals(3, resultRepo.noticeList.size());
+        assertEquals(3, noticeList.size());
         underTest.execute();
-        assertEquals(6, resultRepo.noticeList.size());
+        assertEquals(6, noticeList.size());
         underTest.execute();
-        assertEquals(9, resultRepo.noticeList.size());
+        assertEquals(9, noticeList.size());
     }
 
     @Test
     void providerErrorShouldGenerateNotice() {
 
-        MockResultRepo resultRepo = new MockResultRepo();
+        ValidationResultRepository mockResultRepo = mockResultRepository();
+        GtfsSpecRepository mockSpecRepo = mockSpecRepository();
+        RawFileRepository mockFileRepo = mockFileRepository();
 
         ParseSingleRowForFile underTest = new ParseSingleRowForFile(
                 RawFileInfo.builder().filename("test_empty.tst").build(),
-                new MockRawFileRepo(),
-                new MockSpecRepo(),
-                resultRepo
+                mockFileRepo,
+                mockSpecRepo,
+                mockResultRepo
         );
 
         underTest.execute();
-        assertEquals(1, resultRepo.noticeList.size());
-        Notice notice = resultRepo.noticeList.get(0);
+        assertEquals(1, noticeList.size());
+        Notice notice = noticeList.get(0);
         assertThat(notice, instanceOf(CannotConstructDataProviderNotice.class));
         assertEquals("E002", notice.getId());
         assertEquals("Data provider error", notice.getTitle());
         assertEquals("test_empty.tst", notice.getFilename());
         assertEquals("An error occurred while trying to access raw data for file: test_empty.tst", notice.getDescription());
     }
 
-    private static class MockEntityParser implements GtfsSpecRepository.RawEntityParser {
+    private ValidationResultRepository mockResultRepository() {
+        ValidationResultRepository mockResultRepo =  mock(ValidationResultRepository.class);
+        when(mockResultRepo.addNotice(any(ErrorNotice.class))).thenAnswer(new Answer<Notice>() {
+            public ErrorNotice answer(InvocationOnMock invocation) {
+                ErrorNotice errorNotice = invocation.getArgument(0);
+                noticeList.add(errorNotice);
+                return errorNotice;
+            }
+        });
+
+        return mockResultRepo;
+    }
+
+    private GtfsSpecRepository mockSpecRepository() {
```

Comment:
Naming scheme: reading client code is a bit confusing as these methods almost look like a constructor. I'd suggest using `buildMockResultRepository` or similar naming to ease reading.

Reply:
Fixed